core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -256,7 +256,14 @@ object SparkEnv extends Logging {
       if (rpcEnv.isInstanceOf[AkkaRpcEnv]) {
         rpcEnv.asInstanceOf[AkkaRpcEnv].actorSystem
       } else {
-        val actorSystemPort = if (port == 0) 0 else rpcEnv.address.port + 1
```

Comment:
@markhamstra we can make this a real oneliner if we use Option :)
```
val actorSystemPort = if (port == 0) 0 else Option(rpcEnv.address).map(_.port + 1).getOrElse(port)
```
not sure if that's easier or harder to read. Either way is fine.